stdlib/Test/src/Test.jl

```diff
@@ -1345,7 +1345,10 @@ function testset_forloop(args, testloop, source)
         # they can be handled properly by `finally` lowering.
         if !first_iteration
             pop_testset()
+            finish_errored = true
```

Comment:
Maybe 
```suggestion
            finish_threw = true
```
is more precise?